woocommerce/wc-admin
# Try: Add Lerna to split out utility packages

Fixes #508 – This exposes new libraries, `wc.currency` and `wc.date` for third party developer use. It also sets up these packages in a lerna framework, so we can publish to npm `@woocommerce/currency` and `@woocommerce/date`.

In this PR:

- Moved date & currency to new `packages` directory, can now be included as `@woocommerce/date`, `@woocommerce/currency`
- Added build process for building packages, generates build & build-module like gutenberg packages. Will also generate CSS, but neither package currently has styles.
- NPM scripts:
  - Added `clean` command to wipe all built files from the directory
  - `build` command has been renamed to `build:core`, while `build` itself runs both `build:core` and `build:packages`.
  - `start` command has also been updated to watch packages along with `client` code changes.
  - If you want to watch just the packages for changes, you can run `dev:packages` directly.
  - `publish:check`, `publish:dev`, `publish:prod`: these are lerna commands to build and publish the packages to npm
- Each package now has a `package.json`, `.npmrc`, and `README.md`. See the [Managing Packages workflow from gutenberg](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md#managing-packages) for details.
- The Jest tests have been updated to point to these packages when `@woocommerce/date` etc are found. Further packages will need to be added to `setup-globals.js` as they're created.
- The Webpack config has also been updated to handle `@woocommerce/` packages, and will also need updates as more packages are added.

I've published alpha versions of [date](https://www.npmjs.com/package/@woocommerce/date) & [currency](https://www.npmjs.com/package/@woocommerce/currency) to npm, using `lerna publish --canary`

**To test**

- Build the app, make sure everything succeeds
- There should be no change in functionality
- Check that `wc.currency` and `wc.date` are available on the global object by trying it in the console.
- Check that tests pass (run yourself, not just relying on travis)
- Try out the various npm scripts, though I would avoid publishing packages just yet 🙂
